### PR TITLE
Implementing TDD approach

### PR DESCRIPTION
**Pull request by Othniel2471**
In this branch, we added the tests for the Solver class the added the Sover class later. We implemented the TDD approach.

### CODE REVIEW

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Othniel2471 @tonnytech , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failing linters, kindly fix them

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require_relative '../solver'
+
+describe Solver do
+  describe '#factorial' do
+    it 'accepts n as a single' do
+      solver = Solver.new
+      expect(solver.factorial(1)).to eq 1
+    end
+
```
Comment (iambenkis) [inline spec/solver_spec.rb:9]: - [ ] Keep Solving issues 👍

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi Team (@Othniel2471 , @Munsa1 )

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  All tests are passing

## Suggested changes

- [ ] Kindly fix the broken link attached to the MIT license. It will make your README more professional 👍 

- [ ] I will suggest you to include in your README consistent data to make it credible. I've noticed that in the usage page you included in your file a `list_test_case` which is not in the project. Kindly update that 👍 

Capture


_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by aamirkhan2478 — APPROVED** [review]
Hi @Othniel2471 & @tonnytech, 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewer :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Othniel2471** [comment]
thank you

**Pull request merged**